biocypher/_get.py

```diff
@@ -80,74 +101,79 @@ def __init__(self, cache_dir: Optional[str] = None) -> None:
-            resource (Resource): The resource to download.
-
+            download (Resource or APIRequest): The resource or API request to download.
         Returns:
-            str or list: The path or paths to the downloaded resource(s).
+            list[str]: The path or paths to the downloaded resource(s) or API request(s).
         """
-        expired = self._is_cache_expired(resource)
+        expired = self._is_cache_expired(download)
 
         if expired or not cache:
-            self._delete_expired_resource_cache(resource)
-            logger.info(f"Asking for download of {resource.name}.")
-            paths = self._download_resource(cache, resource)
+            self._delete_expired_cache(download)
+            if isinstance(download, Resource):
+                logger.info(f"Asking for download of resource {download.name}.")
+                paths = self._download_resource(cache, download)
+            else:
+                logger.info(
+                    f"Asking for download of api request {download.name}."
+                )
+                paths = self._download_api_request(download)
         else:
-            paths = self.get_cached_version(resource)
-        self._update_cache_record(resource)
+            paths = self.get_cached_version(download)
+        self._update_cache_record(download)
         return paths
 
-    def _is_cache_expired(self, resource: Resource) -> bool:
+    def _is_cache_expired(self, download: Union[Resource, APIRequest]) -> bool:
         """
-        Check if resource cache is expired.
+        Check if resource or API request cache is expired.
 
         Args:
-            resource (Resource): The resource to download.
+            download (Resource or APIRequest): The resource or API request to download.
 
         Returns:
             bool: cache is expired or not.
         """
-        cache_record = self._get_cache_record(resource)
+        cache_record = self._get_cache_record(download)
         if cache_record:
             download_time = datetime.strptime(
                 cache_record.get("date_downloaded"), "%Y-%m-%d %H:%M:%S.%f"
             )
-            lifetime = timedelta(days=resource.lifetime)
+            lifetime = timedelta(days=download.lifetime)
             expired = download_time + lifetime < datetime.now()
         else:
             expired = True
         return expired
 
-    def _delete_expired_resource_cache(self, resource: Resource):
-        resource_cache_path = self.cache_dir + "/" + resource.name
-        if os.path.exists(resource_cache_path) and os.path.isdir(
-            resource_cache_path
-        ):
-            shutil.rmtree(resource_cache_path)
+    def _delete_expired_cache(self, download: Union[Resource, APIRequest]):
+        cache_path = self.cache_dir + "/" + download.name
+        if os.path.exists(cache_path) and os.path.isdir(cache_path):
+            shutil.rmtree(cache_path)
 
     def _download_resource(self, cache, resource):
```

Comment:
Types should be hinted here, not clear if resource is only Resource or also APIRequest